apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/parser/sdl/GraphSDLSchemaParser.kt

```diff
@@ -106,6 +109,218 @@ object GraphSDLSchemaParser {
     )
   }
 
+  private fun List<GraphSDLParser.TypeSystemExtensionContext>.parse(
+      typeDefinitions: MutableMap<String, GraphSdlSchema.TypeDefinition>
+  ) {
+    forEach {
+      it.typeExtension()?.apply {
+        scalarTypeExtensionDefinition()?.parse(typeDefinitions)
+        enumTypeExtensionDefinition()?.parse(typeDefinitions)
+        objectTypeExtensionDefinition()?.parse(typeDefinitions)
+        interfaceTypeExtensionDefinition()?.parse(typeDefinitions)
+        unionTypeExtensionDefinition()?.parse(typeDefinitions)
+        inputObjectTypeExtensionDefinition()?.parse(typeDefinitions)
+      }
+    }
+  }
+
+  private fun GraphSDLParser.ScalarTypeExtensionDefinitionContext.parse(typeDefinitions: MutableMap<String, GraphSdlSchema.TypeDefinition>) {
+    val name = name().text
+    val scalar = typeDefinitions.get(name)
+    if (scalar == null) {
+      throw ParseException(
+          message = "Cannot add scalar type extension on unknown scalar `$name`",
+          token = start
+      )
+    }
```

Comment:
nit:  just a preference
```
val scalar = typeDefinitions.get(name) ?: throw ParseException(
          message = "Cannot add scalar type extension on unknown scalar `$name`",
          token = start
      )
```

Reply:
IntelliJ agrees with you and put yellow underlines. I replaced with them.